crates/proof-of-sql/src/base/polynomial/interpolate.rs

```diff
@@ -13,7 +13,14 @@ use num_traits::{Inv, One, Zero};
 ///
 /// For any polynomial, `f(x)`, with degree less than or equal to `d`, we have that:
 /// `f(x) = sum_{i=0}^{d} (-1)^(d-i) * (f(i) / (i! * (d-i)! * (x-i))) * prod_{i=0}^{d} (x-i)`
-/// unless x is one of 0,1,...,d, in which case, f(x) is already known.
+/// unless x is one of 0,1,...,d, in which case, f(x) is already known.\
+/// 
+///  # Panics
+/// * This function panics if any of the following occurs:
+///   * When computing the inverse of `(i! * (d-i)! * (x-i))`, if the value `x - i` is zero
+///     (which is guaranteed to not happen as `i == x` will return early).
+///   * If the slice of polynomial coefficients is empty, the function will return `F::zero()`
+///     without performing any operations.
```

Comment:
This isn't a panic, and is normal behavior. The `0` polynomial is always `0` at every point.

Although if this confused you, it appears that this clarifying note should be added to the documentation (outside of the Panic documentation).